### PR TITLE
Ben/add metrics to in mem finetune

### PR DESCRIPTION
**Pull request by benleetownsend**

### CODE REVIEW

File: finetune/util/metrics.py [hunk 1]
```diff
@@ -142,6 +142,20 @@ def sequences_overlap(true_seq, pred_seq):
...
+    f1s = dict()
+    for key in precisions.keys():
+        assert key in recalls
+        p = precisions[key]
+        r = recalls[key]
+        if p == 0 and r == 0:
+            f1s[key] = 0.0
+        else:
+            f1s[key] = 2 * p * r / (p + r)
+    return sum(f1s.values()) / len(f1s)
```
Comment (lhz1029) [inline finetune/util/metrics.py:158]: We might want to consider excluding keys where p and r are 0? it seems like it would artificially skew our f1 down

**Review by lhz1029 — APPROVED** [review]

File: finetune/util/metrics.py (hunk 1 above)
Comment (madisonmay) [reply finetune/util/metrics.py:158]: I don't think it's the place of a utility function like this to make that kind of decision.

Comment (madisonmay) [reply finetune/util/metrics.py:158]: Slash we could artificially remove rare classes as a pre-processing step if we were concerned about this

Comment (madisonmay) [reply finetune/util/metrics.py:158]: A secondary PR to track micro F1 could make a lot of sense though.

File: finetune/util/in_memory_finetune.py [hunk 2]
```diff
@@ -69,9 +72,10 @@ def _evaluate(self, session):
             os.makedirs(directory)
         summary_writer = writer_cache.FileWriterCache.get(directory)
         summary_proto = summary_pb2.Summary()
-        summary_proto.value.add(tag="finetuning/{}_train_accurary".format(self._name), simple_value=float(train_accuracy))
-        summary_proto.value.add(tag="finetuning/{}_test_accurary".format(self._name), simple_value=float(test_accuracy))
+        summary_proto.value.add(tag="finetuning/{}_train_f1".format(self._name), simple_value=float(train_f1))
```
Comment (madisonmay) [inline finetune/util/in_memory_finetune.py:75]: Would be preferable to conditionally compute metrics based on task type but for now I don't think we care.  LGTM

**Review by madisonmay — APPROVED** [review]

**Pull request merged**